protocols/virtualfund/virtualfund.go

```diff
@@ -326,15 +326,24 @@ func (connection *Connection) ledgerChannelAffordsExpectedGuarantees() bool {
 func (s VirtualFundObjective) generateLedgerRequestSideEffects() protocols.SideEffects {
 	sideEffects := protocols.SideEffects{}
 	sideEffects.LedgerRequests = make([]protocols.LedgerRequest, 0)
+
+	leftAmount := s.V.PreFundState().Outcome.TotalAllocatedFor(s.V.MyDestination())
+	// TODO: This is hacky way of getting the second expected outcome.
+	other := s.V.PreFundState().Outcome[0].Allocations[1].Destination
```

Comment:
I think an elegant longer-term solution to this would be to define a type like so

```go
type VirtualChannel struct {
    Channel 
}
```

as we do with two party ledger channels:

https://github.com/statechannels/go-nitro/blob/c35aba4b207fc7cf6436160e872d140e64c0538f/channel/channel.go#L31-L47
 Then, we can have a constructor for this type which does runtime checks on the number of participants and the structure of the outcome, and write methods for `VirtualChannel` such as `V.GetLeftAmount` and `V.GetRightAmount`, similar to this: 

https://github.com/statechannels/go-nitro/blob/c35aba4b207fc7cf6436160e872d140e64c0538f/channel/channel.go#L93-L96
 
 Pinging @NiloCK for his thoughts.

Reply:
I would definitely be in favour of something like this!

Another solution is modifying the `Outcome` struct so that you can get `types.Funds` for an index. Similar to [TotalAllocatedFor](https://github.com/statechannels/go-nitro/blob/ag/multiple-asset/channel/state/outcome/exit.go#L90:L90) but it would accept an index. This would be a less elegant but quicker solution to the TODO